src/VolumeDrawable.ts

```diff
@@ -169,33 +169,47 @@ export default class VolumeDrawable {
       if (hasIso !== options.isosurfaceEnabled) {
         if (hasIso && !options.isosurfaceEnabled) {
           this.destroyIsosurface(channelIndex);
-        } else if (!hasIso && options.isosurfaceEnabled) {
-          // 127 is half of the intensity range 0..255
-          let isovalue = 127;
-          if (options.isovalue !== undefined) {
-            isovalue = options.isovalue;
-          }
-          // 1.0 is fully opaque
-          let isosurfaceOpacity = 1.0;
-          if (options.isosurfaceOpacity !== undefined) {
-            isosurfaceOpacity = options.isosurfaceOpacity;
-          }
-          this.createIsosurface(channelIndex, isovalue, isosurfaceOpacity, isosurfaceOpacity < 1.0);
+        } else if (!hasIso && options.isosurfaceEnabled && this.volume.channels[channelIndex].loaded) {
+          const { isovalue, isosurfaceOpacity } = options;
+          this.createIsosurface(channelIndex, isovalue, isosurfaceOpacity);
         }
+        this.updateChannelDataRequired(channelIndex);
       } else if (options.isosurfaceEnabled) {
         if (options.isovalue !== undefined) {
           this.updateIsovalue(channelIndex, options.isovalue);
         }
         if (options.isosurfaceOpacity !== undefined) {
-          this.updateOpacity(channelIndex, options.isosurfaceOpacity);
+          this.updateIsosurfaceOpacity(channelIndex, options.isosurfaceOpacity);
         }
       }
     } else {
       if (options.isovalue !== undefined) {
         this.updateIsovalue(channelIndex, options.isovalue);
       }
       if (options.isosurfaceOpacity !== undefined) {
-        this.updateOpacity(channelIndex, options.isosurfaceOpacity);
+        this.updateIsosurfaceOpacity(channelIndex, options.isosurfaceOpacity);
+      }
+    }
+  }
+
+  /**
+   * Updates whether a channel's data must be loaded for rendering, based on if its volume or isosurface is enabled.
+   * Calls `Volume.updateRequiredData` to update the list of required channels if necessary.
+   */
+  private updateChannelDataRequired(channelIndex: number): void {
+    const { enabled, isosurfaceEnabled } = this.channelOptions[channelIndex];
+    const channelIsRequired = enabled || isosurfaceEnabled;
+    const requiredChannels = this.volume.loadSpecRequired.channels;
+
+    if (requiredChannels.includes(channelIndex)) {
+      if (!channelIsRequired) {
+        // This channel is currently marked required, but both its volume and isosurface are off. Remove it!
+        this.volume.updateRequiredData({ channels: requiredChannels.filter((i) => i !== channelIndex) });
+      }
+    } else {
+      if (channelIsRequired) {
+        // This channel is not marked required, but either its volume or isosurface is on. Add it!
+        this.volume.updateRequiredData({ channels: [...requiredChannels, channelIndex] });
       }
     }
   }
```

Comment:
I know this is a simple check but it's very readable! I did something similar in another PR and now I want to go back and restructure it to look like this.